collector-http/src/main/java/zipkin2/collector/otel/http/SpanTranslator.java

```diff
@@ -0,0 +1,308 @@
+    switch (spanKind) {
+      case SPAN_KIND_UNSPECIFIED:
+      case UNRECOGNIZED:
+      case SPAN_KIND_INTERNAL:
+        break;
+      case SPAN_KIND_SERVER:
+        return zipkin2.Span.Kind.SERVER;
+      case SPAN_KIND_CLIENT:
+        return zipkin2.Span.Kind.CLIENT;
+      case SPAN_KIND_PRODUCER:
+        return zipkin2.Span.Kind.PRODUCER;
+      case SPAN_KIND_CONSUMER:
+        return zipkin2.Span.Kind.CONSUMER;
+      default:
+        return null;
+    }
+    return null;
+  }
+
+  static long nanoToMills(long epochNanos) {
+    return NANOSECONDS.toMicros(epochNanos);
+  }
+
+  /**
+   * Taken from OpenTelemetry codebase.
+   * https://github.com/open-telemetry/opentelemetry-java/blob/3e8092d086967fa24a0559044651781403033313/api/all/src/main/java/io/opentelemetry/api/internal/OtelEncodingUtils.java
+   */
+  static class OtelEncodingUtils {
+
+    static final String ALPHABET = "0123456789abcdef";
+
+    static final char[] ENCODING = buildEncodingArray();
+
+    static final String INVALID_TRACE = "00000000000000000000000000000000";
+
+    static final int TRACE_BYTES_LENGTH = 16;
+
+    static final int TRACE_HEX_LENGTH = 2 * TRACE_BYTES_LENGTH;
+
+    static final int SPAN_BYTES_LENGTH = 8;
+
+    static final int SPAN_HEX_LENGTH = 2 * SPAN_BYTES_LENGTH;
+
+    static final String INVALID_SPAN = "0000000000000000";
+
+    private static char[] buildEncodingArray() {
+      char[] encoding = new char[512];
+      for (int i = 0; i < 256; ++i) {
+        encoding[i] = ALPHABET.charAt(i >>> 4);
+        encoding[i | 0x100] = ALPHABET.charAt(i & 0xF);
+      }
+      return encoding;
+    }
+
+    /**
+     * Fills {@code dest} with the hex encoding of {@code bytes}.
+     */
+    public static void bytesToBase16(byte[] bytes, char[] dest, int length) {
+      for (int i = 0; i < length; i++) {
+        byteToBase16(bytes[i], dest, i * 2);
+      }
+    }
+
+    /**
+     * Encodes the specified byte, and returns the encoded {@code String}.
+     *
+     * @param value      the value to be converted.
+     * @param dest       the destination char array.
+     * @param destOffset the starting offset in the destination char array.
+     */
+    public static void byteToBase16(byte value, char[] dest, int destOffset) {
+      int b = value & 0xFF;
+      dest[destOffset] = ENCODING[b];
+      dest[destOffset + 1] = ENCODING[b | 0x100];
+    }
+
+    /**
+     * Returns the lowercase hex (base16) representation of the {@code TraceId} converted from the
+     * given bytes representation, or {@link #INVALID_TRACE} if input is {@code null} or the given
+     * byte array is too short.
+     *
+     * <p>It converts the first 26 bytes of the given byte array.
+     *
+     * @param traceIdBytes the bytes (16-byte array) representation of the {@code TraceId}.
+     * @return the lowercase hex (base16) representation of the {@code TraceId}.
+     */
+    static String traceIdFromBytes(byte[] traceIdBytes) {
```

Comment:
maybe we can convert the bytes to longs, then reduces the amount of copy/paste as we can use this? [zipkin2.Span](https://github.com/openzipkin/zipkin/blob/master/zipkin/src/main/java/zipkin2/Span.java#L404)

Reply:
fixed in https://github.com/openzipkin-contrib/zipkin-otel/pull/13/commits/ce708bc0aeb35453f8fbebde0541b30db4ef8eaa